capstone/client/src/components/ListPage.js

```diff
@@ -115,7 +122,28 @@ class ListPage extends Component {
    * @TODO Change this function to make a GET request to obtain store recommendations based 
    * on the selected items.
    */
-  onSubmit = () => {
+   onSubmit = async () => {
+    var latit = this.state.location.latitude;
+    var longi = this.state.location.longitude;
+    if(this.state.zipCodeRequired && (!this.state.zipCode || this.state.zipCode.length !== 5 || isNaN(this.state.zipCode))){
```

Comment:
The check isn't good enough. Zip code 1.111 will pass. Using a regular expression to check is better, such as:

String zipCodeRegex = '/^\d{5}$/'.
return zipCodeRegex.test(input);